fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/FabricEntityTeleporter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.dimension;
+
+import net.fabricmc.fabric.impl.entity.TeleportingServerPlayerEntity;
+import net.minecraft.client.network.packet.*;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.effect.StatusEffectInstance;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.level.LevelProperties;
+
+/**
+ * Entity teleporter that ignores vanilla placement actions (such as nether portals being built).
+ */
+public class FabricEntityTeleporter {
```

Comment:
This class is seriously Forge level of incompatibility. Not only is it extremely vulnerable to any mojang change, it can break mods assumptions. 2 of my mods have mixins that would not be called when people use this method, so I think globally quite a few mods could break.

Reply:
Yeah, having an entirely seperate dimension-switch-logic is not ideal. I would probably have set some custom nbt value and then replaced only vanilla's placement logic with whatever is specified by said nbt value, while keeping everyhting else. Still feels like a very hacky solution though, but I can't come up with anything better right now.